charts/internal/alicloud-infra/templates/main.tf

```diff
@@ -33,6 +33,25 @@ resource "alicloud_vswitch" "vsw_z{{ $index }}" {
   availability_zone = "{{ required "zone.name is required" $zone.name }}"
 }
 
+{{ if $zone.natGateway }}
+{{ if $zone.natGateway.ipAddresses -}}
+// Use specified EIP
+data "alicloud_eip" "eip_natgw_ds{{ $index }}" {
+  ip_addresses = "[{{ index $zone.natGateway.ipAddresses 0 }}]"
```

Comment:
Why is
```yaml
  natGateway:
    ipAddresses:
    - 10.11.12.13
```

a list when you only access index `0`? Why not looping here? If it's not possible, I suggest to change the `values.yaml` structure from `list` to `string`.

Reply:
Thanks for your detailed review on this change.  Currently, alicloud only supports binding multi eips on one vswitch only when `shared bandwidth package` service has been enabled, so customers cannot bind multi eips without `shared bandwidth package` service. The reason why I use `list` is for the consideration of future change adaptation if alicloud allow binding multi eips without `shared bandwidth package` service in the future. In that case, we don't need to change the data structure, and do some change on `main.tf` is ok.

Of course, currently because only one eip can be bound without `shared bandwidth package` service, we cannot use `string` rather than `list`. 

Which one is preferable?

Reply:
OK, thanks for the clarification. If you already know that it'll be possible in the future to bind multiple EIPs to it then it makes sense to keep the list. If that's just a hope/wish without concrete plans from Alicloud's side then I would use a `string`. :)